src/libs/intents/setBiometryState.ts

```diff
@@ -40,6 +42,11 @@ export const openSettingBiometry = async (): Promise<boolean> => {
 export const makeFlagshipMetadataInjection = async (): Promise<string> => {
   const rnBiometrics = new ReactNativeBiometrics()
 
+  const sensorData = await rnBiometrics.isSensorAvailable()
+  const biometryDenied = BIOMETRY_DENIED_BY_USER_IOS_ERROR.test(
+    sensorData.error ?? ''
+  )
```

Comment:
Why don't you use the `isBiometryDenied` you created in the next commit?

Reply:
At first I used the Metadata injection as it was not supposed to change during the app's lifecycle (when changing OS settings, the app restarts automatically)
But then I realized that on the first launch, the authorization would change this prop's state without restarting the app. So I implemented the new "on demand" method.

I hesitated to refactor the first commit and also use the `onDemand`. This would imply calling an async method on cozy-settings side to initialize the component state
![image](https://user-images.githubusercontent.com/1884255/200590010-d6d0f38e-c5e9-4c44-b09c-4309e224e62a.png)
Nothing difficult here. I'm wondering if this would introduce some flickering (but maybe it's a false assumption). I'll try it.

Reply:
Yep it was just about reusing the method. I thought at first that the code was exactly the same.